feat(turbo-ignore): args and refactor (#2671)
vercel/turbo: packages/turbo-ignore/README.md

This uses `turbo` to automatically determine if the current app has new changes that need to be deployed.

## Usage

Use `npx turbo-ignore --help` to see list of options:

```sh
turbo-ignore

Automatically ignore builds that have no changes

Usage:
  $ npx turbo-ignore [<workspace>] [flags...]

If <workspace> is not provided, it will be inferred from the "name"
field of the "package.json" located at the current working directory.

Flags:

>   --filter-fallback   If no previously deployed SHA is available to compare with,

Reviewer: Should this explicitly say `VERCEL_DEPLOY_SHA`?
```suggestion
  --filter-fallback   If `VERCEL_DEPLOY_SHA` is not set,
```

>   --filter-fallback   If no previously deployed SHA is available to compare with,

Reviewer: Don't love the `--filter-fallback` naming. Can't think of an obviously better name, but something like `--compare-to-previous-sha` or `--ignore-env-var` or something?

Author: Yea me either. This is _only_ used when the `VERCEL_GIT_PREVIOUS_SHA` isn't there, so it should be treated as a fallback and not skip or ignore.

                      fallback to comparing against the previous commit [default: true]
  --help, -h          Show this help message
  --version, -v       Show the version of this script
```

## How it Works

> `turbo-ignore` determines if a build should continue by analyzing the dependency graph of the given workspace.

Reviewer: ```suggestion
`turbo-ignore` determines if a build should continue by analyzing the package dependency graph of the given workspace.
```

The _given workspace_ is determined by reading the "name" field in the "package.json" file located at the current working directory, or by passing in a workspace name as the first argument to `turbo-ignore`.

> Next, it used `turbo` to determine if the given workspace, _or any dependencies of the workspace_, have changed since the previous commit.

Reviewer: ```suggestion
Next, it uses `turbo run build --dry` to determine if the given workspace, _or any dependencies of the workspace_, have changed since the previous commit.
```

> **NOTE:** `turbo` determines dependencies from reading the dependency graph of the given workspace. This means a dependency **must** be listed as a dependency in the given workspaces `package.json` for `turbo` to recognize it as a dependency.

Reviewer: Specify `dependencies` vs `devDependencies` if relevant?
